src/backends/wayland/mod.rs

```diff
@@ -136,7 +136,12 @@ impl<D: HasDisplayHandle + ?Sized, W: HasWindowHandle> WaylandImpl<D, W> {
             self.surface().commit();
         }
 
-        let _ = self.display.event_queue.borrow_mut().flush();
+        let _ = self
```

Comment:
Maybe instead of repeating this pattern in a few different places, `DisplayImpl` could have a `lock_event_queue` method?

Otherwise the Wayland changes look good.

Reply:
I disagree. This pattern is only repeated three times, is around five lines and I'm worried that a `lock_event_queue` method would obscure the point of the code more than it would help.